test/TestVersioned.py

```diff
@@ -61,4 +61,46 @@ def test_Custom_Value_Fill(self, requests):
 
         query = requests.getQs()
         self.assertEqual(query["fillerK"],"CustomValue")
-        self.assertEqual(query["fillerDV"],"10")
\ No newline at end of file
+        self.assertEqual(query["fillerDV"],"10")
+
+    @helpers.TrackRequests
+    def test_ForMostRecent(self, requests):
+        url = qs.createVersioned() \
+            .forMarketData([100000001]) \
+            .inAbsoluteDateRange("2018-01-01","2018-01-02") \
+            .inTimeZone("UTC") \
+            .inGranularity(Granularity.HOUR) \
+            .forMostRecent("2018-01-01","2018-01-02") \
+            .withFillCustomValue(10) \
+            .execute()
+
+        query = requests.getQs()
+        self.assertEqual(query["fillerDV"],"10")
+
+    @helpers.TrackRequests
+    def test_ForMostRecentDateTime(self, requests):
+        url = qs.createVersioned() \
+            .forMarketData([100000001]) \
+            .inAbsoluteDateRange("2021-09-22","2021-09-23") \
+            .inTimeZone("CET") \
+            .inGranularity(Granularity.DAY) \
+            .forMostRecent("2021-09-22T12:30:05","2021-09-23T00:00:00") \
+            .withFillCustomValue(10) \
+            .execute()
+
+        query = requests.getQs()
+        self.assertEqual(query["fillerDV"],"10")
+
+    @helpers.TrackRequests
+    def test_ForMostRecentDateTimeFillNull(self, requests):
+        url = qs.createVersioned() \
+            .forMarketData([100000001]) \
+            .inAbsoluteDateRange("2021-09-22","2021-09-23") \
+            .inTimeZone("CET") \
+            .inGranularity(Granularity.DAY) \
+            .forMostRecent("2021-09-22T12:30:05","2021-09-23T00:00:00") \
+            .withFillNull() \
+            .execute()
+
+        query = requests.getQs()
+        self.assertEqual(query["fillerK"],"Null")
```

Comment:
asserts are not checking on the MostRecent part of the query (the url)

am I missing something?

Reply:
I think I'm missing something! I don't know how to build the query (I figure that's the URL). I tried adding ForMostRecent with/without dates etc. but I'm kind of shooting in the dark. I'm not that familiar with Python. Plus I don't know how to run the tests in debug mode so I can't observe the query to get an idea of how to modify the query.

Reply:
Me neither, but I'm sure someone in your team does, as someone did this code.

The point is that you added tests related to updating the MostRecent with something, not clear from PR. I know that is related to the time part but I don't see any checks that the created url is as expected.

You added test case but you didn't tested them?